libbeat/publisher/queue/memqueue/eventloop.go

```diff
@@ -112,61 +157,309 @@ func (l *eventLoop) handleCancel(req *producerCancelRequest) {
 
 	if st := req.state; st != nil {
 		st.cancelled = true
-		removed = broker.cancel(st)
+		removed = l.buf.cancel(st)
 	}
 
 	// signal cancel request being finished
 	if req.resp != nil {
-		req.resp <- producerCancelResponse{
-			removed: removed,
-		}
+		req.resp <- producerCancelResponse{removed: removed}
 	}
 
 	// re-enable pushRequest if buffer can take new events
-	if !broker.full() {
+	if !l.buf.Full() {
 		l.events = broker.events
 	}
 }
 
-func (l *eventLoop) handleConsumer(req *getRequest) {
+func (l *directEventLoop) handleConsumer(req *getRequest) {
 	// log := l.broker.logger
+	// log.Debugf("try reserve %v events", req.sz)
 
-	start, buf := l.broker.get(req.sz)
+	start, buf := l.buf.reserve(req.sz)
 	count := len(buf)
 	if count == 0 {
 		panic("empty batch returned")
 	}
 
 	// log.Debug("newACKChan: ", b.ackSeq, count)
-	ackCH := newACKChan(l.ackSeq, start, count)
+	ackCH := newACKChan(l.ackSeq, start, count, l.buf.buf.clients)
```

Comment:
Naming smells a bit here ;-)

Reply:
it's due to the old code. Some more cleanups will come at some point.